.github/workflows/pr-build.yml

```diff
@@ -61,8 +63,42 @@ jobs:
           flags: integration
 
       - name: push docker image
-        if: github.event.pull_request.merged
+         # https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#pull_request
+        if: github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged == true
```

Comment:
Discussed offline with @JBAhire , rather than getting fancy with the conditions and triggers which seems more fragile, we can just split the workflows up to not share triggers - the extra job is justified IMO (explained more below)